# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/rene/vnc/missing-powerpc-templates.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


This patch adds the missing PowerPC definitions (am I / are we the only
person / people on earth that build for PowerPC ?!?!?!?!?!?!?!?)

  - Rene Rebe <rene@rocklinux.org>

--- vnc_unixsrc/Xvnc/config/cf/Imake.cf.orig	2003-09-06 03:01:25.000000000 +0200
+++ vnc_unixsrc/Xvnc/config/cf/Imake.cf	2003-09-06 03:01:13.000000000 +0200
@@ -552,6 +552,10 @@
 #  define AlphaArchitecture
 #  undef __alpha
 # endif /* __alpha */
+# ifdef __powerpc
+#  define ppcArchitecture
+#  undef __powerpc
+# endif /* __powerpc */
 # ifdef mc68000
 #  define Mc68020Architecture
 #  undef mc68000
--- vnc_unixsrc/Xvnc/config/cf/linux.cf.orig	1998-05-11 18:38:06.000000000 +0200
+++ vnc_unixsrc/Xvnc/config/cf/linux.cf	2003-09-06 02:51:52.000000000 +0200
@@ -197,6 +197,13 @@
 #define AsmDefines		-D__ELF__
 #define CplusplusCmd		c++ -b alpha-linux
 #endif /* AlphaArchitecture */
+#ifdef ppcArchitecture
+#define CcCmd                   gcc
+#define AsCmd                   /usr/powerpc-linux/bin/as
+#define LdCmd                   ld -m elf32powerpc
+#define AsmDefines              -D__ELF__
+#define CplusplusCmd            c++ -b powerpc-linux
+#endif /* ppcArchitecture */
 #ifdef i386Architecture
 #define CcCmd			gcc -b i486-linux
 #define AsCmd			/usr/i486-linux/bin/as
@@ -252,6 +259,13 @@
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
 #endif /* AlphaArchitecture */
 
+#ifdef ppcArchitecture
+#define OptimizedCDebugFlags    -O2
+#define LinuxMachineDefines     -D__powerpc__
+#define ServerOSDefines         XFree86ServerOSDefines -DDDXTIME -DPART_NET
+#define ServerExtraDefines      -DGCCUSESGAS XFree86ServerDefines
+#endif /* ppcArchitecture */
+
 #ifdef Mc68020Architecture
 #define OptimizedCDebugFlags	-O2
 #define LinuxMachineDefines 	-D__mc68000__
